text/0778-instrumentation-using-debug-render-tree.md

```diff
@@ -0,0 +1,109 @@
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Instrumentation using Debug-Render-Tree
+
+## Summary
+
+> Currently, instrumation for Ember component has payload with view data, which includes BOUNDS. This enables the render highlight feature for Ember-Inspector. We can also use it for tracking component rendering. Unfortunately, this is not supported by Glimmer component, the payload for Glimmer component only contains component name. However, debug-render-tree is fully supported by Glimmer component, they contains view data including BOUNDS. We can integrate instrumentation with debug-render-tree then we can send view data to Ember-Inspector. 
+
+## Motivation
+
+> First of all, Ember component supports such feature. The payload data can be used by some other apps. However, after Octane migration, this is no longer supported. It is an unexpected deprecation. I saw several requests for such feature to be continuously supported 
+> https://discord.com/channels/480462759797063690/501388188762505216/684442737113563213
+> https://discord.com/channels/480462759797063690/501388188762505216/686516553050751005
+> Additionally, I recently added the component render highlight feature for Ember-Inspector. 
+> https://drive.google.com/file/d/1jIUPaM69okID1tlroAFu8lead6zeKfi_/view?usp=sharing
+> Other framework like ReactJS has such feature in dev-tool for several years. It is very important for developers, since we can then see when the component is updated or created. It not only help us improve the website performance by eliminating wasted rendering but also help detects potential bugs when component is destroyed and recreated. Without instrumenation with view data, these are impossible.
+
+
+## Detailed design
+### This is what we currently have:
+> When a Glimmer component is rendered. "Bounds" data is send to debugRenderTree
+> https://github.com/glimmerjs/glimmer-vm/blob/master/packages/@glimmer/runtime/lib/compiled/opcodes/component.ts#L841
+```js
+vm.env.debugRenderTree!.didRender(bucket, bounds);
+```
+> debugRenderTree save the data in the map `nodes`
+> https://github.com/glimmerjs/glimmer-vm/blob/master/packages/@glimmer/runtime/lib/debug-render-tree.ts#L86
+```js
+  didRender(state: TBucket, bounds: Bounds): void {
+    ...
+
+    this.nodeFor(state).bounds = bounds;
+    this.exit();
+  }
+```
+> Currently, we only use the `capture` method to construct the render tree
+> https://github.com/emberjs/ember.js/blob/master/packages/@ember/debug/lib/capture-render-tree.ts#L25
+```js
+export default function captureRenderTree(app: Owner): CapturedRenderNode[] {
+  let renderer = expect(app.lookup<Renderer>('renderer:-dom'), `BUG: owner is missing renderer`);
+
+  return renderer.debugRenderTree.capture();
+}
+```
+### My proposal:
+> We can add an array `newNodes` with `state` element. We push `state` to queue
+```js
```

Comment:
```suggestion

```js
```

Question: do you have a markdown linter in your IDE by any chance? It would help you auto-format your document more easily (or at least give you feedback on formatting markdown). 👍